node.gyp

```diff
@@ -780,6 +791,14 @@
         'common.gypi',
       ],
 
+      'conditions': [
+        [ 'node_shared_uvwasi=="false"', {
+          'sources': [
+#          'src/node_wasi.cc',
+#          'src/node_wasi.h'
```

Comment:
@PoojaDurgad I think removing these two files is why you have the build failure. `src/node_wasi.cc` registers the builtin wasi module: https://github.com/nodejs/node/blob/757d7ba12b156ee2c5aaca1702bab2f95d28c750/src/node_wasi.cc#L1743
which ends up creating `_register_wasi()`:
https://github.com/nodejs/node/blob/a3d8de956ed5043d9bbbc09d30ed9d5b1e54186b/src/node_binding.h#L47-L48
https://github.com/nodejs/node/blob/a3d8de956ed5043d9bbbc09d30ed9d5b1e54186b/src/node_binding.h#L27-L38